prolog/st/st_tokens.pl

```diff
@@ -36,29 +60,22 @@
 
 tokens([], _) --> "".
 
-comment(simple) -->
-    "{{%", string(_), "}}", !.
-
-comment(semblance) -->
-    "{#", string(_), "#}", !.
-
-out(simple, out(Term)) -->
-    "{{=", whites, term_to_token(Term, `}}`), !.
-
-out(semblance, out(Term)) -->
-    "{{", whites, term_to_token(Term, `}}`), !.
+comment(syntax_tokens(comment(Start, End), out(_,_), out_unescaped(_,_), statement(_,_))) -->
+    Start, string(_), End, !.
 
-out_unescaped(simple, out_unescaped(Term)) -->
-    "{{-", whites, term_to_token(Term, `}}`), !.
+out(syntax_tokens(comment(_,_), out(Start, End), out_unescaped(_,_), statement(_,_)), out(Term)) -->
+    { string_codes(End, EndCodes) },
+    Start, whites, term_to_token(Term, EndCodes), !.
 
-out_unescaped(semblance, out_unescaped(Term)) -->
-    "{%", whites, "unescape", whites, term_to_token(Term, `%}`), !.
+out_unescaped(syntax_tokens(comment(_,_), out(_,_), out_unescaped(Start, End), statement(_,_)), out_unescaped(Term)) -->
+    { string_codes(End, EndCodes) },
+    Start, whites, term_to_token(Term, EndCodes), !.
 
 token(Term, Frontend) -->
-    out(Frontend, Term), !.
+    out_unescaped(Frontend, Term), !.
 
 token(Term, Frontend) -->
-    out_unescaped(Frontend, Term), !.
+    out(Frontend, Term), !.
```

Comment:
Swapped the order of these two DCG rules so that `unescape` keywords don't get gobbled by `out`. This prefers `out_unescaped` when encountered.